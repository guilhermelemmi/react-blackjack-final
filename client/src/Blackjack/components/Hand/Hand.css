.hand {
  display: inline-flex;
  overflow-x: auto;
  width: 400px;
  margin-right: 30px;
}

.hand div {
  margin-left: -30%;
}

.hand div:first-child {
  margin-left: 0;
}

.hand div:last-child {
  flex-shrink: 0;
}

@media only screen and (max-width: 690px) {
  .hand {
    width: 230px;
    margin-right: 10px;
  }
}
