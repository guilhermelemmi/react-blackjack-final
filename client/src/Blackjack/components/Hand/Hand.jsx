import "./Hand.css";

function Hand({ children }) {
  return (
    <div className="hand" >
      {children}
    </div>
  );
}

export default Hand;
