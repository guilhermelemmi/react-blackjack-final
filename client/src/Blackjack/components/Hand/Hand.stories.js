import Card from "../Card/Card";
import Hand from "./Hand";

export default {
  component: Hand,
};

export const Blackjack = {
  render: () => (
    <Hand>
      <Card key="1_1" suit={1} rank={1} />
      <Card key="1_13" suit={1} rank={13} />
    </Hand>
  ),
};

export const FourAces = {
  render: () => (
    <Hand>
      <Card key="1_1" suit={1} rank={1} />
      <Card key="1_2" suit={2} rank={1} />
      <Card key="1_3" suit={3} rank={1} />
      <Card key="1_4" suit={4} rank={1} />
    </Hand>
  ),
};
