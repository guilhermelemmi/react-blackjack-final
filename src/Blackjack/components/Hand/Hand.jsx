import "./Hand.css";

function Hand({ children, ...otherProps }) {
  return (
    <div className="hand" {...otherProps} >
      {children}
    </div>
  );
}

export default Hand;
