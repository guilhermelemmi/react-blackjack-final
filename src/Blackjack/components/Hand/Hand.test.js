import { render, screen } from "@testing-library/react";
import Hand from "./Hand";

describe("Hand", () => {
  test("It renders its children", async () => {
    render(
      <Hand>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Hand>,
    );
    const div1 = screen.getByText("1");
    const div2 = screen.getByText("2");
    const div3 = screen.getByText("3");
    expect(div1).toBeInTheDocument();
    expect(div2).toBeInTheDocument();
    expect(div3).toBeInTheDocument();
  });

  test("It matches snapshot", async () => {
    const view = render(
      <Hand>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Hand>,
    );
    expect(view.container).toMatchSnapshot();
  });
});
