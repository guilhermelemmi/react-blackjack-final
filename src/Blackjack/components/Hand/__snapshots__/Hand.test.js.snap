// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hand It matches snapshot 1`] = `
<div>
  <div
    class="hand"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div>
      3
    </div>
  </div>
</div>
`;
